core/embed/rust/src/ui/model_mercury/flow/confirm_summary.rs

```diff
@@ -105,34 +110,42 @@ pub fn new_confirm_summary(
     });
 
     // FeeInfo
-    let has_fee_info = !fee_params.is_empty();
-    let content_fee = fee_params.into_layout()?;
-
+    let content_fee = fee_params.map(|params| params.into_layout()).transpose()?;
     // AccountInfo
-    let has_account_info = !account_params.is_empty();
-    let content_account = account_params.into_layout()?;
+    let content_account = account_params
+        .map(|params| params.into_layout())
+        .transpose()?;
+    // ExtraInfo
+    let content_extra = extra_params
+        .map(|params| params.into_layout())
+        .transpose()?;
 
-    // Menu
+    // Menu with provided info and cancel
     let mut menu = VerticalMenu::empty();
     let mut menu_items = Vec::<usize, 3>::new();
-    if has_fee_info {
+    if content_fee.is_some() {
         menu = menu.item(
             theme::ICON_CHEVRON_RIGHT,
             TR::confirm_total__title_fee.into(),
         );
         unwrap!(menu_items.push(MENU_ITEM_FEE_INFO));
     }
-    if has_account_info {
+    if content_account.is_some() {
         menu = menu.item(
             theme::ICON_CHEVRON_RIGHT,
             TR::address_details__account_info.into(),
         );
         unwrap!(menu_items.push(MENU_ITEM_ACCOUNT_INFO));
     }
-    menu = menu.danger(
-        theme::ICON_CANCEL,
-        cancel_text.unwrap_or(TR::send__cancel_sign.into()),
-    );
+    if content_extra.is_some() && menu_items.len() < 2 {
```

Comment:
I don't really like the hardcoded "2" here. Ideally it would be somehow computed based on the screen size, but a big step forward (probably good enough) would be a constant in `vertical_menu.rs` - something like `MAX_ITEMS_ON_SCREEN`.

Reply:
29cf6d3557b4c6f877c38fe39c959148df6e4e21